src/client/pages/Embark/index.tsx

```diff
@@ -245,23 +245,28 @@ interface AngelVariables {
 const useInitQuoteCart = ({ skip = false }) => {
   const quoteCartId = useOnboardingQuoteCartId()
   const [addCampaignCode] = useAddCampaignCodeMutation()
+  const { search } = useLocation()
+  const searchParams = useMemo(() => new URLSearchParams(search), [search])
 
   const initQuoteCart = useCallback(async () => {
+    const campaignCodeFromUrl = searchParams.get('campaignCode')
     const savedCampaignCode = CampaignCode.get()
 
-    if (savedCampaignCode !== null) {
+    const campaignCode = campaignCodeFromUrl ?? savedCampaignCode
+
+    if (campaignCode !== null) {
       try {
         await addCampaignCode({
           variables: {
             id: quoteCartId,
-            code: savedCampaignCode,
+            code: campaignCode,
           },
         })
       } catch {
         CampaignCode.remove()
       }
     }
-  }, [addCampaignCode, quoteCartId])
+  }, [addCampaignCode, quoteCartId, searchParams])
 
   useEffect(() => {
```

Comment:
Nit: you can apply same idea here.  Since effect runs once, you can move initQuoteCart declaration inside it and remove one more hook.  Unrelated to your change, I know.

Reply:
Nice, will fix this as well  👌